src/test/java/com/wikia/webdriver/common/testnglisteners/InvokeMethodAdapter.java

```diff
@@ -53,11 +59,58 @@ private boolean isTestExcludedFromEnv(Method method) {
       String[] excludedEnvs = method.getAnnotation(DontRun.class).env();
 
       for (String excludedEnv : excludedEnvs) {
-        if (Configuration.getEnv().contains(excludedEnv)) {
+        if (Configuration.getEnv().equals(excludedEnv)) {
           return true;
         }
       }
     }
+    if (method.isAnnotationPresent(RunOnly.class)) {
+      String[] includedEnvs = method.getAnnotation(RunOnly.class).env();
+
+      if (includedEnvs.length == 1 && includedEnvs[0].isEmpty()) {
+        return false;
+      }
+
+      for (String includedEnv : includedEnvs) {
+        if (Configuration.getEnv().equals(includedEnv)) {
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Returns true if test is excluded from running with current language
+   */
+  private boolean isTestExcludedFromLang(Method method) {
+    if (method.isAnnotationPresent(DontRun.class)) {
+      String[] excludedLangs = method.getAnnotation(DontRun.class).language();
+
+      String langFromConfig = System.getProperty("language") != null ? System.getProperty("language")
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "language" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)